### PR TITLE
Make RPC test output more deterministic

### PR DESCRIPTION
**Pull request by sellout**
* sort the result summary (always)
* add a flag, `--deterministic` (`-d`), to enable other output changes
* don’t print progress `.` (with `-d`)
* don’t print durations/runtime (with `-d`)

This isn’t fully deterministic as tests can still complete out of order, but even there these changes make it easier to diff regions.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

File: qa/pull-tester/rpc-tests.py [hunk 1]
```diff
@@ -185,6 +185,7 @@ def main():
     Help text and arguments for individual test script:''',
                                      formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument('--coverage', action='store_true', help='generate a basic coverage report for the RPC interface')
+    parser.add_argument('--deterministic', '-d', action='store_true', help='make the output a bit more deterministic in order to compare runs.')
```
Comment (daira) [inline qa/pull-tester/rpc-tests.py:188]:
```suggestion
    parser.add_argument('--deterministic', '-d', action='store_true', help='make the output a bit closer to deterministic in order to compare runs.')
```

**Comment by sellout** [comment]
I force-pushed what should have been a separate commit. It has two fixes:
1. use `success_rate` instead of `boolean` for summary even when not using `--deterministic` and
2. fix the `exit` status, which had broken when I stopped updating `all_passed`.

**Comment by sellout** [comment]
I think this does give deterministic output if you do
```bash
./qa/pull-tester/rpc-tests.py --deterministic --jobs 1
```
but ideally it would still give deterministic output without sacrificing parallelization.

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**